From 97efa086ee7598ba80e7e27f64e443fe8f5bc0bf Mon Sep 17 00:00:00 2001
From: Climate Corporation <daad-embedded-software@bayer.com>
Date: Wed, 5 Jun 2019 15:03:09 -0500
Subject: [PATCH] Do not add newline to B64-encoded data

---
 src/cencode.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/cencode.c b/src/cencode.c
index 03ba5b6..e21a7b5 100644
--- a/src/cencode.c
+++ b/src/cencode.c
@@ -75,7 +75,7 @@ int base64_encode_block(const char* plaintext_in, int length_in, char* code_out,
 			++(state_in->stepcount);
 			if (state_in->stepcount == CHARS_PER_LINE/4)
 			{
-				*codechar++ = '\n';
+//Do not add newline		*codechar++ = '\n';
 				state_in->stepcount = 0;
 			}
 		}
@@ -102,7 +102,7 @@ int base64_encode_blockend(char* code_out, base64_encodestate* state_in)
 	case step_A:
 		break;
 	}
-	*codechar++ = '\n';
+// Do not add newline	*codechar++ = '\n';
 	
 	return codechar - code_out;
 }
-- 
2.7.4

